scripts/test_screeps_tencent_batch_rl_runner.py

```
@@ -0,0 +1,180 @@
+                    runner.safe_extract_tar(tar_path, root / "remote")
+
+    def test_scale_down_raises_after_recording_failed_modify_desired_capacity(self) -> None:
+        with tempfile.TemporaryDirectory() as temp_dir:
+            controller = runner.Controller(args=controller_args(), run_id="run-test", artifact_dir=Path(temp_dir))
+
+            with mock.patch.object(
+                runner.subprocess,
+                "run",
+                return_value=subprocess.CompletedProcess(["tccli"], 1, "", "permission denied"),
+            ):
+                with self.assertRaisesRegex(runner.BatchRunError, "scale_down failed"):
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)